pkg/actors/reminders/reminders.go

```diff
@@ -235,23 +254,34 @@ func (r *reminders) RenameReminder(ctx context.Context, req *internal.RenameRemi
 		return err
 	}
 
+	// Start the new reminder
 	return r.startReminder(reminder, stop)
 }
 
 func (r *reminders) evaluateReminders(ctx context.Context) {
-	r.evaluationLock.Lock()
-	defer r.evaluationLock.Unlock()
-
-	r.evaluationChan <- struct{}{}
+	// Wait for the evaluation channel
+	select {
+	case r.evaluationChan <- struct{}{}:
```

Comment:
Should we have a `default` case as well here which just says `return`?

Reply:
Nope, this needs to block and wait :)